File: Stronghold/Stronghold.Application/DTOs/Request/AdminCreateAppointmentRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Stronghold.Application.DTOs.Request;

public class AdminCreateAppointmentRequest : IValidatableObject
{
    [Required(ErrorMessage = "Korisnik je obavezan.")]
    public int UserId { get; set; }

    public int? TrainerId { get; set; }
    public int? NutritionistId { get; set; }

    [Required(ErrorMessage = "Datum termina je obavezan.")]
    public DateTime AppointmentDate { get; set; }

    public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
    {
        if (TrainerId == null && NutritionistId == null)
            yield return new ValidationResult("Morate odabrati trenera ili nutricionistu.");

        if (TrainerId != null && NutritionistId != null)
            yield return new ValidationResult("Termin moze biti samo kod trenera ili nutricioniste, ne oba.");

        if (AppointmentDate <= DateTime.UtcNow)
            yield return new ValidationResult("Datum termina mora biti u buducnosti.");
    }
}

File: packages/stronghold_core/lib/models/requests/admin_create_appointment_request.dart
import '../common/date_time_utils.dart';

class AdminCreateAppointmentRequest {
  final int userId;
  final int? trainerId;
  final int? nutritionistId;
  final DateTime appointmentDate;

  const AdminCreateAppointmentRequest({
    required this.userId,
    this.trainerId,
    this.nutritionistId,
    required this.appointmentDate,
  });

  Map<String, dynamic> toJson() => {
        'userId': userId,
        if (trainerId != null) 'trainerId': trainerId,
        if (nutritionistId != null) 'nutritionistId': nutritionistId,
        'appointmentDate': DateTimeUtils.toApiDateTime(appointmentDate),
      };
}
